src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -207,32 +207,31 @@ def _wait_for_topics(self, admin_client, topics, timeout=10):
     def create_consumer(
         self,
         topics,
-        bootstrap_servers,
+        cluster_options,
         group_id,
         auto_offset_reset,
         queued_max_messages_kbytes,
         queued_min_messages,
     ):
 
-        consumer_config = {
-            "enable.auto.commit": False,
-            "bootstrap.servers": ",".join(bootstrap_servers),
-            "group.id": group_id,
-            "default.topic.config": {"auto.offset.reset": auto_offset_reset},
-            # overridden to reduce memory usage when there's a large backlog
-            "queued.max.messages.kbytes": queued_max_messages_kbytes,
-            "queued.min.messages": queued_min_messages,
-        }
+        consumer_config = cluster_options.copy()
```

Comment:
I am a bit uncomfortable with these `copy()` spread around where we need to override the config. 
Is there any issue with calling `get_kafka_consumer_cluster_options` here and passing the dictionary of overrides so that that function takes care of doing the copies when needed in a transparent way ?

Reply:
Refactored. Did my best to avoid .copy() calls - had to restructure a bit the Synchronized consumer and tests